File: typescript-examples/playwright-basics-ts/api/examples/scrapeList.ts
import { BrowserContext, Page } from "playwright";

interface Params {
  // Add your params if needed
}

export default async function handler(
  params: Params,
  page: Page,
  context: BrowserContext
) {
  const dataList: Array<{ id: string; name: string; status: string }> = [];

  // Navigate to the page
  await page.goto("https://sandbox.intuned.dev/lists/table/");

  // Locate the main table
  const containerLocator = page.getByRole("table");

  if (await containerLocator.isVisible() && (await containerLocator.count()) > 0) {
    // Get all rows in the table
    const rows = await containerLocator.locator("tbody tr").all();

    for (const row of rows) {
      // Scrape Id
      const idLocator = row.locator("td").nth(0);
      const itemId = (await idLocator.count()) > 0 ? await idLocator.textContent() : null;

      // Scrape Name
      const nameLocator = row.locator("td:nth-of-type(2)");
      const name = (await nameLocator.count()) > 0 ? (await nameLocator.textContent())?.trim() : null;

      // Scrape Status
      const statusLocator = row.locator("td").last();
      const status = (await statusLocator.count()) > 0 ? (await statusLocator.innerText()).trim() : null;

      if (itemId && name && status) {
        dataList.push({ id: itemId, name, status });
      }
    }
  }

  return dataList;
}

File: python-examples/playwright-python/api/examples/scrape-list.py
from playwright.async_api import Page, Locator
from typing import TypedDict


class Params(TypedDict):
    pass


async def automation(page: Page, params: Params | None = None, **_kwargs):
    data_list = []
    # Navigate to your page
    await page.goto("https://sandbox.intuned.dev/lists/table/")

    # Locate the main container to check it's state
    container_locator = page.get_by_role("table")
    if await container_locator.is_visible() and await container_locator.count() > 0:
        # Get all rows in the table
        rows = await container_locator.locator("tbody tr").all()
        # For each row scrape column data and create an object to push it to the list
        for row in rows:
            # Scrape Id
            id_locator = row.locator("td").nth(0)
            item_id = (
                await id_locator.text_content()
                if await id_locator.count() > 0
                else None
            )

            # Scrape Name
            name_locator = row.locator("td:nth-of-type(2)")
            status_locator = row.locator("td").last
            name = (
                (await name_locator.text_content()).strip()
                if await name_locator.count() > 0
                else None
            )

            # Scrape Status
            status = (
                (await status_locator.inner_text()).strip()
                if await status_locator.count() > 0
                else None
            )

            # You Can scrape other fields  here

            if item_id and name and status:
                # Create an object in order to push it to the list
                item = {"id": item_id, "name": name, "status": status}

                data_list.append(item)

    return data_list
